sources/packages/backend/libs/test-utils/src/factories/designation-agreement-locations.ts

```diff
@@ -27,24 +34,35 @@ export async function saveFakeDesignationAgreementLocation(
     ),
   );
   const fakeUser = await db.user.save(createFakeUser());
-  const designationAgreement = await db.designationAgreement.save(
-    createFakeDesignationAgreement({
-      fakeInstitution,
-      fakeInstitutionLocations,
-      fakeUser,
-    }),
+  const savedDesignationAgreement = await db.designationAgreement.save(
+    createFakeDesignationAgreement(
+      {
+        fakeInstitution,
+        fakeInstitutionLocations,
+        fakeUser,
+      },
+      {
+        initialValue: {
+          endDate: options.initialValues?.endDate,
+          assessedDate: options.initialValues?.assessedDate,
+        },
+      },
+    ),
   );
-
-  const designationAgreementLocations = fakeInstitutionLocations.map(
-    (fakeInstitutionLocation) => {
-      const designationAgreementLocation = new DesignationAgreementLocation();
-      designationAgreementLocation.designationAgreement = designationAgreement;
-      designationAgreementLocation.institutionLocation =
-        fakeInstitutionLocation;
-      designationAgreementLocation.requested = true;
-      designationAgreementLocation.approved = true;
-      return designationAgreementLocation;
-    },
+  const designationAgreementLocations =
+    savedDesignationAgreement.designationAgreementLocations;
+  designationAgreementLocations.forEach(
+    (designationAgreementLocation) =>
+      (designationAgreementLocation.approved = false),
+  );
+  const [designatedLocation] = designationAgreementLocations;
+  designatedLocation.approved = true;
+  const savedDesignationAgreementLocations =
+    await db.designationAgreementLocation.save(designationAgreementLocations);
+  savedDesignationAgreementLocations.forEach(
+    (savedDesignationAgreementLocation) =>
+      (savedDesignationAgreementLocation.designationAgreement =
```

Comment:
what are we doing here, can you please explain?

Reply:
By default, the eager is set to false for `DesignationAgreement`

<img width="672" alt="image" src="https://github.com/bcgov/SIMS/assets/7859295/efecd757-4c70-48cb-9d0f-6dfd78ed36e8">

As a result I am explicitly attaching the `designationAgreement` object to the created `designationAgreementLocation` before it is returned as it is needed by the e2e test.